FirebaseDatabaseUI/FirebaseIndexArray.m

```diff
@@ -0,0 +1,205 @@
+// clang-format off
+
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+// clang-format on
+
+#import "FirebaseIndexArray.h"
+#import "FirebaseQueryObserver.h"
+
+@interface FirebaseIndexArray () <FirebaseArrayDelegate>
+
+@property (nonatomic, readonly) id<FIRDataObservable> index;
```

Comment:
Not a big fan of the readonly here - if we're declaring private property decls lets make them readwrite and use self.property when setting.